### PR TITLE
Make oclif commands testable

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/cli/commands/BaseCommand.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+		const user = new User();
+
+		Object.assign(user, { ...this.defaultUserProps, globalRole });
+
+		await Db.collections.User.save(user);
+
+		return Db.collections.User.findOneOrFail({ globalRole });
+	}
+}
```
Comment (ivov) [inline packages/cli/commands/BaseCommand.ts:57]: Reusable in `execute` and `executeBatch` after tests added.

File: packages/cli/src/Db.ts [hunk 2]
```diff
@@ -45,6 +45,8 @@ export function linkRepository<Entity>(entityClass: EntityTarget<Entity>): Repos
 export async function init(
 	testConnectionOptions?: ConnectionOptions,
 ): Promise<IDatabaseCollections> {
+	if (isInitialized) return collections;
```
Comment (ivov) [inline packages/cli/src/Db.ts:48]: During test, prevent command from reinitializing `Db` and so from overwriting test DB connection options with actual DB.

**Review by BHesseldieck — APPROVED** [review]

**Comment by janober** [comment]
Great, thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.184.0`

**Pull request merged**